Long stack improvements

jbunton-atlassian opened the pull request:
When longStackSupport is enabled count each stack trace that is created.

While following the promise chain keep appending stacks as long as they are older than the source.

jbunton-atlassian commented:
Hi @domenic,

I don't suppose you have some time to have a look at this PR?

Thanks :)

abartomeu commented:
Hi @domenic,

Any progress on this? 

Thanks

wmertens commented:
LGTM, in that I can't see this doing harm, but I don't understand how it works :) Maybe there should be a comment regarding how maintaining the counter helps for this feature?

jbunton-atlassian commented:
I have added a comment which hopefully clarifies it.

The old stopping condition was to simply check if the "From previous event" string existed in the stack trace. The test cases I added demonstrate that this is not sufficient to produce useful long stacks in all cases. Removing the stopping condition altogether results in duplicate entries in the long stack and/or entries which simply should not be there.

The counter approach allows q to stop at exactly the correct stack :)

wmertens commented:
Looks great, thanks!!

jbunton-atlassian commented:
Thanks for merging! Is anybody able to publish this as a new version on npmjs.com?

wmertens commented:
@domenic @kriskowal could you push a new version to npm?
